mantidimaging/core/reconstruct/cil_recon.py

```diff
@@ -64,6 +64,8 @@ def __init__(self, verbose=1, residual_interval: int = 1) -> None:
         self.residual_interval = residual_interval
 
     def __call__(self, algo: Algorithm) -> None:
+        if not hasattr(algo, 'last_residual'):
```

Comment:
Would it be neater to put the `hasattr` check go in `MIProgressCallback.__call__()` on line 49?

Reply:
as in `if hasattr(algo, 'last_residual') and algo.last_residual[0] == algo.iteration:`?
or
```python
if not hasattr(algo, "last_residual"):
    algo.last_residual = None
if algo.last_residual and algo.last_residual[0] == algo.iteration:
    extra_info["residual"] = algo.last_residual[1]
```

Reply:
The first one

Reply:
yeah, I agree. just rebased now. thanks for noticing.